### DISCUSSION TITLE
sam.predict with BBOX :  No valid pixel coordinates found.  TypeError: 'NoneType' object is not subscriptable

### DISCUSSION THREAD

**Discussion by alex6H** [opengeos/segment-geospatial#270] [Q&A] [answered]
Hello,

First of all thanks a lot for this great lib !

I have an issue with SAM and the bbox segmentation. I followed the tutorial [here](https://samgeo.gishub.org/examples/box_prompts/) which run perfectly on my notebook. When I use my data which are from a .shp I have a error "No valid pixel coordinates found." 
TypeError: 'NoneType' object is not subscriptable

Here is my code :
```
# Initiate
sam = SamGeo(
    model_type="vit_b",
    automatic=False,
    device='cuda',
    sam_kwargs=None,
)

# Load Geotiff
sam.set_image(SOURCE_IMAGE_PATH)

# Read .shp
gdf_mask = gpd.read_file("output.shp").to_crs(4326)

# Create bbox
gdf_bboxes = gdf_mask['geometry'].bounds
bbox_list = gdf_bboxes[['minx','miny','maxx','maxy']].to_numpy().tolist()
bbox_list


# Use box prompts
sam.predict(
            boxes= bbox_list,
            point_crs="EPSG:4326",
            output="mask2.tif", 
            dtype="uint8",
           )
)
```

Here is the error: 

> No valid pixel coordinates found.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[78], line 2
      1 # Use box prompts
----> 2 sam.predict(
      3             boxes= bbox_list,
      4             point_crs="EPSG:3857",
      5             output="mask2.tif", 
      6             dtype="uint8",
      7            )

File [~\Anaconda3\envs\Yolo8\lib\site-packages\samgeo\samgeo.py:559](http://localhost:8888/lab/tree/OneDrive%20-%20/~/Anaconda3/envs/Yolo8/lib/site-packages/samgeo/samgeo.py#line=558), in SamGeo.predict(self, point_coords, point_labels, boxes, point_crs, mask_input, multimask_output, return_logits, output, index, mask_multiplier, dtype, return_results, **kwargs)
    557 coords = bbox_to_xy(self.source, boxes, point_crs)
    558 input_boxes = np.array(coords)
--> 559 if isinstance(coords[0], int):
    560     input_boxes = input_boxes[None, :]
    561 else:

TypeError: 'NoneType' object is not subscriptable

My bbox_list look like this: 
[[29.12381803879782,
    -1.611772506698625,
    29.124026117545267,
    -1.611600683403621],
   [29.123034288297987,
    -1.6135127229398778,
    29.12307628216605,
    -1.6134663678379761]]
![image](https://github.com/opengeos/segment-geospatial/assets/48403935/4f87a1c3-638a-45c1-882a-82341ee1cd81)

I have tried different format but can't find a way to have a segmentation.

Can you help me ?

Thanks !

**Comment by giswqs** [comment]
What is the crs of the image?

**Reply by alex6H** [reply]
Also EPSG:4326

![image](https://github.com/opengeos/segment-geospatial/assets/48403935/17aa8435-f33e-4955-83b9-3625a6e8e772)


When I open both (.shp and .tif) in QGIS they do overlap and ESPG are the same
![image](https://github.com/opengeos/segment-geospatial/assets/48403935/a65b8058-2804-46b0-bd47-4b12dfe5c043)

**Reply by alex6H** [reply]
I forgot to mention that my .tif is not a RGB but an orthomosaic create from a set of .jpg (from drone images)

**Reply by giswqs** [reply]
Try to reproject the image so that it has a projection, such as `EPSG:3857`

**Reply by alex6H** [reply, marked as answer]
You are my hero !

Solution : 
raster in EPSG:3857
.shp stays in EPSG:4326

then : 
```
# Use box prompts
sam.predict(
            boxes= bbox_list,
            point_crs="EPSG:4326",
            output="mask2.tif", 
            dtype="uint8",
           )
```

**Reply by alex6H** [reply]
Thank you so much !

**Reply by giswqs** [reply]
Glad to hear that it worked.
